File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& arr) {
        int l,r,sum =0;
        int ans =INT_MAX;
        while(r<arr.size()){
            sum +=arr[r];
            while(sum>=target){
                ans = min(ans,r-l+1);
                sum-=arr[l++];
            }
            r++;
        }
        return ans==INT_MAX?0:ans;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l =0, r=0, sum =0;
        int ans  = Integer.MAX_VALUE;
        while(r<nums.length){
            sum+=nums[r];
            while(sum>=target){
                ans  = Math.min(ans,r-l+1);
                sum-=nums[l++];
            }
            r++;
        }
        return ans == Integer.MAX_VALUE?0:ans;
    }
}